### PR TITLE
Rework the whole code structure.

### PR DESCRIPTION
**Pull request by armic-artelys**
Changed the code structure : used the base ideas from Bertrand with a more generic approach.

### CODE REVIEW

File: src/main/java/com/powsybl/integrationtest/loadflow/jsonconfig/LoadflowTestPlanReader.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
+package com.powsybl.integrationtest.loadflow.jsonconfig;
```
Comment (miovd) [inline src/main/java/com/powsybl/integrationtest/loadflow/jsonconfig/LoadflowTestPlanReader.java:1]: Add copyright

File: src/main/java/com/powsybl/integrationtest/loadflow/model/LoadflowComputationParameters.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package com.powsybl.integrationtest.loadflow.model;
```
Comment (miovd) [inline src/main/java/com/powsybl/integrationtest/loadflow/model/LoadflowComputationParameters.java:1]: Add copyright

File: src/main/java/com/powsybl/integrationtest/loadflow/model/LoadflowComputationResults.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package com.powsybl.integrationtest.loadflow.model;
```
Comment (miovd) [inline src/main/java/com/powsybl/integrationtest/loadflow/model/LoadflowComputationResults.java:1]: Add copyright

File: src/main/java/com/powsybl/integrationtest/loadflow/model/LoadflowComputationResults.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+import com.powsybl.iidm.network.Network;
+import com.powsybl.integrationtest.model.ComputationResults;
+import com.powsybl.loadflow.LoadFlowResult;
+
+/**
+ *
```
Comment (miovd) [inline src/main/java/com/powsybl/integrationtest/loadflow/model/LoadflowComputationResults.java:8]: Add author

File: src/main/java/com/powsybl/integrationtest/loadflow/model/LoadflowTestCase.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package com.powsybl.integrationtest.loadflow.model;
```
Comment (miovd) [inline src/main/java/com/powsybl/integrationtest/loadflow/model/LoadflowTestCase.java:1]: Add copyright

File: src/main/java/com/powsybl/integrationtest/loadflow/model/LoadflowTestCase.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+import com.powsybl.integrationtest.model.TestCase;
+
+public class LoadflowTestCase implements TestCase<LoadflowComputationParameters, LoadflowComputationResults> {
```
Comment (miovd) [inline src/main/java/com/powsybl/integrationtest/loadflow/model/LoadflowTestCase.java:5]: Add author

File: src/main/java/com/powsybl/integrationtest/model/TestCase.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
+package com.powsybl.integrationtest.model;
```
Comment (miovd) [inline src/main/java/com/powsybl/integrationtest/model/TestCase.java:1]: Add copyright

File: src/main/java/com/powsybl/integrationtest/model/TestPlan.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
+package com.powsybl.integrationtest.model;
```
Comment (miovd) [inline src/main/java/com/powsybl/integrationtest/model/TestPlan.java:1]: Add copyright

File: src/main/java/com/powsybl/integrationtest/model/TestRunner.java [hunk 9]
```diff
@@ -0,0 +1,21 @@
+package com.powsybl.integrationtest.model;
```
Comment (miovd) [inline src/main/java/com/powsybl/integrationtest/model/TestRunner.java:1]: Add copyright

File: src/test/java/com/powsybl/integrationtest/LoadflowIntegrationTest.java [hunk 10]
```diff
@@ -0,0 +1,36 @@
+package com.powsybl.integrationtest;
```
Comment (miovd) [inline src/test/java/com/powsybl/integrationtest/LoadflowIntegrationTest.java:1]: Add copyright

File: src/test/java/com/powsybl/integrationtest/LoadflowIntegrationTest.java [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+import com.powsybl.integrationtest.model.TestPlan;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringJoiner;
+
+public class LoadflowIntegrationTest {
```
Comment (miovd) [inline src/test/java/com/powsybl/integrationtest/LoadflowIntegrationTest.java:17]: Add author

File: src/main/java/com/powsybl/integrationtest/loadflow/jsonconfig/LoadflowTestCaseJson.java [hunk 12]
```diff
@@ -17,54 +17,54 @@
...
+    public LoadflowTestCaseJson(String id, String inputNetwork, String inputParameters, String expectedNetwork, String expectedResults) {
         this.id = id;
-        this.network = network;
-        this.parameters = parameters;
-        this.networkRef = networkRef;
-        this.resultRef = resultRef;
+        this.inputNetwork = inputNetwork;
+        this.inputParameters = inputParameters;
+        this.expectedNetwork = expectedNetwork;
+        this.expectedResults = expectedResults;
```
Comment (miovd) [inline src/main/java/com/powsybl/integrationtest/loadflow/jsonconfig/LoadflowTestCaseJson.java:37]: Should we add `Objects.requireNonNull`? (Same remark for all public constructors)

**Review by miovd — COMMENTED** [review]
Thank you for your work, I am okay with the new code structure. I only have some remarks about copyrights, authorship and checking null attributes on public constructors.

**Comment by armic-artelys** [comment]
OK thanks @miovd, I will quickly work on the small changes to add and come back with a commit.

**Comment by armic-artelys** [comment]
- [x] Added copyright.
- [x] Fixed files authorship.
- [x] Improved robustness by checking for parameters nullity in public constructors.

**Review by obrix — APPROVED** [review]

**Review by miovd — APPROVED** [review]

**Pull request merged**